src/pages/products/Products.js

```diff
@@ -8,18 +13,21 @@ class Products extends Component {
     originListProd: [],
     listProd: []
   }
-  
 
   static getDerivedStateFromProps(nextProps, prevState) {
-    if (nextProps.listProd !== prevState.originListProd) {
+    if (nextProps.products !== prevState.originListProd) {
```

Comment:
do not overuse `getDerivedStateFromProps`. In most cases it is possible to solve tasks without this method.